pupil-spa/src/app/services/check-complete/check-complete.service.ts

```diff
@@ -0,0 +1,105 @@
+import { APP_CONFIG } from '../config/config.service';
+import { AuditService } from '../audit/audit.service';
+import { AzureQueueService } from '../azure-queue/azure-queue.service';
+import {
+  CheckSubmissionApiCalled,
+  CheckSubmissionAPIFailed,
+  CheckSubmissionAPICallSucceeded,
+} from '../audit/auditEntry';
+import { Injectable } from '@angular/core';
+import { Router } from '@angular/router';
+import { SubmissionService } from '../submission/submission.service';
+import { StorageService } from '../storage/storage.service';
+import { TokenService } from '../token/token.service';
+
+/**
+ * Declaration of check start service
+ */
+@Injectable()
+export class CheckCompleteService {
+
+  checkSubmissionApiErrorDelay;
+  checkSubmissionAPIErrorMaxAttempts;
+  featureUseHpa;
+  submissionPendingViewMinDisplay;
+
+  constructor(private auditService: AuditService,
+              private azureQueueService: AzureQueueService,
+              private router: Router,
+              private storageService: StorageService,
+              private submissionService: SubmissionService,
+              private tokenService: TokenService) {
+    const { featureUseHpa,
+      checkSubmissionApiErrorDelay,
+      checkSubmissionAPIErrorMaxAttempts,
+      submissionPendingViewMinDisplay,
+    } = APP_CONFIG;
+    this.checkSubmissionApiErrorDelay = checkSubmissionApiErrorDelay;
+    this.checkSubmissionAPIErrorMaxAttempts = checkSubmissionAPIErrorMaxAttempts;
+    this.featureUseHpa = featureUseHpa;
+    this.submissionPendingViewMinDisplay = submissionPendingViewMinDisplay;
+  }
+
+  /**
+   * Sleep function (milliseconds) to provide minimal display time for submission pending screen
+   * @param {Number} ms
+   * @returns {Promise.<void>}
+   */
+  private sleep(ms) {
+    return new Promise(resolve => setTimeout(resolve, ms));
+  }
+
+  /**
+   * Check complete submission
+   * @param {Number} startTime Date time in milliseconds on the exact moment before check submission is called
+   * @returns {Promise.<void>}
+   */
+  public async submit(startTime): Promise<void> {
+    if (this.featureUseHpa === true) {
+      const queueName = 'check-complete';
+      const {url, token} = this.tokenService.getToken('checkComplete');
+      const retryConfig = {
+        errorDelay: this.checkSubmissionApiErrorDelay,
+        errorMaxAttempts: this.checkSubmissionAPIErrorMaxAttempts
+      };
+      this.auditService.addEntry(new CheckSubmissionApiCalled());
+      const payload = this.storageService.getAllItems();
+      try {
+        await this.azureQueueService.addMessage(queueName, url, token, payload, retryConfig);
+        this.auditService.addEntry(new CheckSubmissionAPICallSucceeded());
+        await this.onSuccess(startTime);
```

Comment:
Does this need `await`-ing?

Reply:
@danwilliams `onSuccess` calls `sleep` which does return a Promise.

Reply:
Okay, after conversation it seems this is deliberate in order to implement a definite and required wait before returning. Happy with it due to that 👍